android/ui/src/main/java/com/trafi/ui/Badge.kt

```diff
@@ -0,0 +1,258 @@
+package com.trafi.ui
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.AmbientContentColor
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.trafi.ui.BadgeType.MEDIUM_BADGE
+import com.trafi.ui.BadgeType.SMALL_BADGE
+import com.trafi.ui.component.internal.BadgeConstants
+import com.trafi.ui.theme.currentTheme
+
+@Composable
+private val constants
+    get() = BadgeConstants(currentTheme)
+
+data class BadgeInfo(
+    val text: String?,
+    val color: Color,
+    val textColor: Color? = null
+)
+
+@OptIn(ExperimentalLayout::class)
+@Composable
+fun Badge(
+    modifier: Modifier = Modifier,
+    badge: BadgeInfo?,
+    alternativeBadges: List<BadgeInfo> = listOf(),
+    vector: ImageVector? = null,
+    subbadge: ImageVector? = null,
+    badgeType: BadgeType,
```

Comment:
Could this have a reasonable default value, e.g. `Medium`? Also interesting why you decided to split `BadgeType` from `BadgeInfo`.